torch/testing/_internal/common_optimizers.py

```diff
@@ -497,6 +495,16 @@ def optim_inputs_func_adam(device, dtype=None):
             kwargs={"lr": torch.tensor(0.001), "amsgrad": True, "capturable": True},
             desc="Tensor lr with capturable and amsgrad",
         ),
+        OptimizerInput(
+            params=None,
+            kwargs={
+                "lr": torch.tensor(0.001),
+                "betas": (torch.tensor(0.9), torch.tensor(0.99)),
+                "amsgrad": True,
+                "capturable": True,
+            },
+            desc="Tensor lr, Tensor betas, with capturable and amsgrad",
+        ),
```

Comment:
Have not reviewed the rest of PR, but these would be good to address along with CI:
We should also add 3 more variations where:
- capturable is False and amsgrad is False
- capturable is True and amsgrad is False
- capturable is False and amsgrad is True

There should either be a test where only 1 beta is a Tensor if that's supported or a test that it errors nicely if not.

Reply:
SG, I'll add those cases. I'll error nicely, don't think it's worth supporting 1 constant and one tensor

Reply:
Note I also banned the capturable is False case (like we did for LR)